### PR TITLE
[JENKINS-37757] Protection against NullPointerException

### PR DESCRIPTION
**Pull request by amuniz**
[JENKINS-37757](https://issues.jenkins-ci.org/browse/JENKINS-37757)

I've not been able to diagnose what is the root cause of the NPE I'm seeing, but this check suppress them.

@reviewbybees

### CODE REVIEW

**Comment by rsandell** [comment]
:bee:

**Comment by varmenise** [comment]
🐝

**Comment by amuniz** [comment]
@reviewbybees done

**Comment by stephenc** [comment]
:bee: 👍

**Comment by recena** [comment]
:+1:

**Comment by amuniz** [comment]
This is the corresponding PR on XStream project: https://github.com/x-stream/xstream/pull/65

**Comment by oleg-nenashev** [comment]
👍 
Warning for project contributors: In order to retain your sanity, please do not launch static analyzer against this component

File: xstream/src/java/com/thoughtworks/xstream/core/DefaultConverterLookup.java [hunk 1]
```diff
@@ -71,7 +73,8 @@ public synchronized void registerConverter(Converter converter, int priority) {
         converters.add(converter, priority);
         for (Iterator iter = typeToConverterMap.keySet().iterator(); iter.hasNext();) {
             Class type = (Class) iter.next();
-            if (converter.canConvert(type)) {
+            if ((converter instanceof NullConverter && converter.canConvert(type)) 
+                    || (type != null && converter.canConvert(type))) {
```
Comment (batmat) [inline xstream/src/java/com/thoughtworks/xstream/core/DefaultConverterLookup.java]: NIT: now the condition got longer, maybe factor out that in a private method?

**Comment by batmat** [comment]
:+1:

**Comment by ndeloof** [comment]
@amuniz did you considered to propose same fix to https://github.com/x-stream/xstream ?

**Comment by amuniz** [comment]
@ndeloof I did: https://github.com/x-stream/xstream/pull/65, but it is under review.

**Comment by batmat** [comment]
This PR should probably be also applied to https://github.com/jenkinsci/xstream-fork as this repo is deprecated

**Comment by oleg-nenashev** [comment]
@batmat @ndeloof I have no idea why it is deprected.  From what I see @ndeloof did that, but I see no conversations in the Jenkins Dev mailing list or anywhere else.

**Comment by oleg-nenashev** [comment]
@amuniz This patch has never been released. Are you still interested in it?

**Pull request merged**